components/brave_wallet_ui/common/reducers/wallet_reducer.ts

```diff
@@ -114,4 +118,52 @@ reducer.on(WalletActions.setAllTokensList, (state: any, payload: GetAllTokensRet
   }
 })
 
+reducer.on(WalletActions.ethBalancesUpdated, (state: any, payload: GetBalanceReturnInfo[]) => {
+  let accounts: WalletAccountType[] = [...state.accounts]
+
+  accounts.forEach((account, index) => {
+    if (payload[index].success) {
+      accounts[index].balance = formatBalance(payload[index].balance, 18)
+      accounts[index].fiatBalance = formatFiatBalance(payload[index].balance, 18, '2000').toString()  // TODO: use actual price info
+    }
+  })
+
+  return {
+    ...state,
+    accounts
+  }
+})
+
+reducer.on(WalletActions.tokenBalancesUpdated, (state: any, payload: GetERC20TokenBalanceReturnInfo[][]) => {
+  const userVisibleTokensInfo: TokenInfo[] = state.userVisibleTokensInfo
+  let accounts: WalletAccountType[] = [...state.accounts]
+
+  accounts.forEach((account, accountIndex) => {
+    payload[accountIndex].forEach((info, tokenIndex) => {
+      let assetBalance = '0'
+      let fiatBalance = '0'
+
+      if (userVisibleTokensInfo[tokenIndex].contractAddress === 'eth') {
+        assetBalance = account.balance
+        fiatBalance = account.fiatBalance
+      } else if (info.success) {
+        assetBalance = formatBalance(info.balance, userVisibleTokensInfo[tokenIndex].decimals)
```

Comment:
Instead of formatting the token balance here, we should keep actual `weiBalance` and only format the balance at the time of displaying it. Since we will need the actual `weiBalance` to make a transaction.

Reply:
Fixed in https://github.com/brave/brave-core/pull/9694/commits/0bd3fccd32f3e7d3d7e34ead0a33a8f1bf2f9757